advanced_alchemy/types/file_object/backends/fsspec.py

```diff
@@ -0,0 +1,404 @@
+
+
+if TYPE_CHECKING:
+    from fsspec import AbstractFileSystem  # pyright: ignore[reportMissingTypeStubs]
+    from fsspec.spec import AbstractBufferedFile  # pyright: ignore[reportMissingTypeStubs]
+
+
+class FSSpecBackend(StorageBackend):
+    """FSSpec-backed storage backend implementing both sync and async operations."""
+
+    driver = "fsspec"  # Changed backend identifier to driver
+    default_expires_in = 3600
+
+    def __init__(
+        self,
+        fs: "Union[AbstractFileSystem, AsyncFileSystem, str]",
+        key: str,
+        *,
+        options: Optional[dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize FSSpecBackend.
+
+        Args:
+            fs: The FSSpec filesystem instance (sync or async).
+            key: The key of the backend instance.
+            options: Optional backend-specific options.
+            **kwargs: Additional keyword arguments.
+        """
+        self.fs = fsspec.filesystem(fs, **options or {}) if isinstance(fs, str) else fs  # pyright: ignore
+        self.is_async = isinstance(self.fs, AsyncFileSystem)
+        protocol = getattr(self.fs, "protocol", None)
+        protocol = cast("Optional[str]", protocol[0] if isinstance(protocol, (list, tuple)) else protocol)
+        self.protocol = protocol or "file"
+        self.key = key
+        self.options = options or {}
+        self.kwargs = kwargs
+
+    def get_content(self, path: PathLike, *, options: Optional[dict[str, Any]] = None) -> bytes:
+        """Return the bytes stored at the specified location.
+
+        Args:
+            path: Path to retrieve
+            options: Optional backend-specific options passed to fsspec's open.
+        """
+        options = options or {}
+        path_str = self._to_path(path)
+        fs = cast("AbstractFileSystem", self.fs)  # Cast for type checker
+        with fs.open(path_str, "rb", **options) as f:  # pyright: ignore
+            # Ensure f is treated as a file handle supporting read
+            # Using AbstractBufferedFile which is common for fsspec file-like objects
+            f_handle = cast("AbstractBufferedFile", f)
+            return f_handle.read()  # type: ignore[no-any-return]
+
+    async def get_content_async(self, path: PathLike, *, options: Optional[dict[str, Any]] = None) -> bytes:
+        """Return the bytes stored at the specified location asynchronously.
+
+        Args:
+            path: Path to retrieve
+            options: Optional backend-specific options passed to fsspec's open.
+
+        """
+        if not self.is_async:
+            # Fallback for sync filesystems - Note: get_content is sync, wrapping with async_
+            return await async_(self.get_content)(path=path, options=options)
+
+        options = options or {}
+        path_str = self._to_path(path)
+        fs = cast("AsyncFileSystem", self.fs)
+        async with fs.open(path_str, mode="rb", **options) as f:  # pyright: ignore
```

Comment:
This doesn't work on async fs. You should do `fs._cat_file()`, which actually does the async read